### DISCUSSION TITLE
opencv4nodejs support end of life ?

### DISCUSSION THREAD

**Discussion by RSM-SBA** [appium/appium#16041] [Q&A] [answered]
I have some old opencv installations on mac minis (which are from before catalina) which are still working after upgrading to BigSur.
Now i had to install openCV from the scratch on a BigSur node, but it doesn't work anymore.
Is here somebody who has expertise on installing opencv4nodejs on BigSur so it is working with appium ?

Would be bad if we have to miss this great feature in appium.

**Comment by jlipps** [comment, marked as answer]
I got it to work at one point. I think I had to specify an older version of opencv, and include some arcane build flags in the build env vars to not try and build certain extraneous parts of opencv. I don't have the steps I followed but looking through my shell history, I see that I used this env var: ` OPENCV4NODEJS_AUTOBUILD_FLAGS="-DBUILD_opencv_text=OFF"`

**Reply by RSM-SBA** [reply]
Thanks, i have tried it with the latest version which does not work.
On Catalina i used
`brew install https://homebrew.bintray.com/bottles/opencv-4.2.0_1.catalina.bottle.tar.gz`
and
` npm install -g opencv4nodejs@5.5.0`
 that worked somehow.

But as bintray is down i have no clue where to find this "old" opencv bottle.

@jlipps which old version did u used ?

**Reply by jlipps** [reply]
I think it was opencv version 3. I also saw an update on this issue, you might try to follow these steps: https://github.com/justadudewhohacks/opencv4nodejs/issues/775#issuecomment-961364849

**Reply by RSM-SBA** [reply]
Thanks, that link helped much. I could build now opencv.

for everyone who has the same problem :
(used node 12)
brew install opencv@3
brew link opencv@3
export OPENCV4NODEJS_DISABLE_AUTOBUILD=0
export OPENCV4NODEJS_AUTOBUILD_FLAGS=-DBUILD_LIST=core,imgproc,imgcodecs,videoio,highgui,video,calib3d,features2d,objdetect,dnn,ml,flann,photo,stitching,gapi

npm i -g opencv-build
npm i -g opencv4nodejs
sudo npm link opencv4nodejs
